### PR TITLE
allow to choose between verbatim and ipv4 first dns resolution order

### LINKED ISSUE #2053: Default DNS lookup fallback to ipv4first
In Node.js 17 the default for DNS lookup was changed from ipv4first to "verbatim". We hard fix this in js-controller 4.0.24 as seen in https://github.com/ioBroker/ioBroker.js-controller/pull/2052

For js-controller 4.1/5 we should do the same but configurable in iobroker.json "system" section. Default should stay for us "ipv4first" but users can change it to "verbatim". The relevant method exists since v16.4.0, v14.18.0.

### PR DESCRIPTION
**Pull request by foxriver76**
- closes #2053

waiting for #2110 before merge (move types to then existing internal package)

### CODE REVIEW

**Comment by foxriver76** [comment]
if we want to we can also publish the schema on schemastore for files matching `iobroker.json` any maybe `iobroker-dist.json` also admin could validate against schema. Thoughts? Or maybe there are expected conflicts, so that other files are also often named `iobroker.json`?

File: packages/common/src/lib/common/tools.ts [hunk 1]
```diff
@@ -3843,4 +3846,20 @@ export function maybeArrayToString<T>(maybeArr: T): T extends any[] ? string : T
...
 
+/**
+ * Ensure that DNS is resolved according to ioBroker config
+ */
+export function ensureDNSOrder(): void {
+    let dnsOrder: 'ipv4first' | 'verbatim' = 'ipv4first';
+    try {
+        const configName = getConfigFileName();
+        const config: ioBroker.IoBrokerJson = fs.readJSONSync(configName);
+        dnsOrder = config.dnsResolution;
```
Comment (Apollon77) [inline packages/common/src/lib/common/tools.ts:3857]: This will break for all existing installations because it will set "undefined"

**Comment by Apollon77** [comment]
maybe then we should alsoi kill nodejs 12 testing at least with this PR?
And I requested changes to  other PR because: >= v16.4.0, v14.18.0

**Review by Apollon77 — CHANGES_REQUESTED** [review]

**Review by Apollon77 — APPROVED** [review]

**Comment by Apollon77** [comment]
Important: Merhing this will break also all nodejs 12 testing ... so we now have Two PRs with this "dep". inform devs and do in some days?

**Pull request merged**